css/css-grid/parsing/grid-template-areas-valid.html

```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>CSS Grid Layout Module Level 1: parsing grid-template-areas with valid values</title>
+<link rel="author" title="Eric Willigers" href="mailto:ericwilligers@chromium.org">
+<link rel="help" href="https://drafts.csswg.org/css-grid/#propdef-grid-template-areas">
+<meta name="assert" content="grid-template-areas supports the full grammar 'none | <string>+'.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/parsing-testcommon.js"></script>
+</head>
+<body>
+<script>
+test_valid_value("grid-template-areas", "none");
+test_valid_value("grid-template-areas", '"first"');
+test_valid_value("grid-template-areas", '"first second"');
+test_valid_value("grid-template-areas", '"1st 2nd 3rd"');
+test_valid_value("grid-template-areas", '"first second" "third fourth"');
+test_valid_value("grid-template-areas", '"first second" "third ." "1st 2nd" "3rd 4th"');
+
+// Firefox preserves specific whitespace and full stop sequences.
+// Other browsers consolidate to ' ' and '.'
```

Comment:
Again spec says:
> Computed value: specified value

Is Firefox right or I'm missing something else.

Reply:
I tried to include a Firefox serialization, but the following error occurs:

```
test_valid_value("grid-template-areas", '"  a  \t  b  "', '" a \\9 b "');
test_valid_value("grid-template-areas", '"  a  \t  b  "', '"  a  \\9  b  "');

assert_equals: serialization should be canonical expected "\" a \\9 b \"" but got "\" a \\9 b \""
assert_equals: serialization should be canonical expected "\" a \\9 b \"" but got "\" a \\9 b \""
```